llvm/lib/Target/RISCV/RISCVInstrInfoVSDPatterns.td

```diff
@@ -907,6 +907,19 @@ foreach vti = AllIntegerVectors in {
   }
 }
 
+// (add v, C) -> (sub v, -C) if -C cheaper to materialize
+foreach vti = I64IntegerVectors in {
+  let Predicates = [HasVInstructionsI64] in {
+    def : Pat<(add (vti.Vector vti.RegClass:$rs1),
+                   (vti.Vector (SplatPat (i64 negImm:$rs2)))),
+//                 (riscv_vmv_v_x_vl undef, negImm:$rs2, srcvalue)),
```

Comment:
This doesn't match the `vadd_vx_imm64_to_sub` test yet (neither `SplatPat` nor `riscv_vmv_v_x_vl`).
I'd appreciate hints for how to debug. Or maybe you can see why it isn't working?
Perhaps I need to add `SplatPat_simmXXX` ? If so, why?

Reply:
I think its because ComplexPatterns can't nest. SplatPat and negImm are both ComplexPatterns.

Reply:
Resolved by adding a combined ComplexPattern.
Pity there's no TableGen diagnostic for nested ComplexPatterns.